app/file_sender.py

```diff
@@ -1,36 +1,109 @@
+import hashlib
+import json
 import logging
+from datetime import datetime
 from pathlib import Path
 from time import sleep
+from typing import Collection, Iterable
 
 from structlog import wrap_logger
 
+import app.sftp as sftp
+from app.encryption import pgp_encrypt_message
+from app.mappings import PRODUCTPACK_CODE_TO_DESCRIPTION, PACK_CODE_TO_DATASET, \
+    SUPPLIER_TO_SFTP_DIRECTORY, DATASET_TO_SUPPLIER
 from config import Config
 
+
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-def process_complete_file(file: Path):
-    # TODO Encrypt, create manifest, send to sftp, delete partial file
-    logger.info(f'doing the file stuff on {file}')
+def process_complete_file(print_file: Path, pack_code):
+    supplier = DATASET_TO_SUPPLIER[PACK_CODE_TO_DATASET[pack_code]]
+
+    logger.info('Encrypting print_file', file_name=print_file.name)
+    encrypted_print_file, filename = encrypt_print_file(print_file, pack_code, supplier)
+
+    manifest_file = Config.ENCRYPTED_FILES_DIRECTORY.joinpath(f'{filename}.manifest')
+    logger.info('Creating manifest print_file', manifest_file=manifest_file.name)
+    generate_manifest_file(manifest_file, encrypted_print_file, pack_code)
+    file_paths = [encrypted_print_file, manifest_file]
+
+    logger.info('Sending files to SFTP', file_paths=file_paths)
+    copy_files_to_sftp(file_paths, SUPPLIER_TO_SFTP_DIRECTORY[supplier])
+
+    # TODO upload encrypted print file and manifest to GCS
```

Comment:
this TODO can go now

Reply:
ah no... what happened to the bucket upload code?

Reply:
The bucket upload wasn't working so we removed it. We figured it could be done as a separate ticket since this PR has grown quite hefty already and is blocking a lot of stuff.